src/serializer/ResidualFunctions.js

```diff
@@ -77,8 +86,12 @@ export class ResidualFunctions {
     );
     this.residualFunctionInfos = residualFunctionInfos;
     this.residualFunctionInstances = residualFunctionInstances;
-    for (let instance of residualFunctionInstances.values())
-      this.addFunctionInstance(((instance: any): FunctionInstance));
+    this.additionalFunctionValueInfos = additionalFunctionValueInfos;
+    for (let instance of residualFunctionInstances.values()) {
+      let functionInstance = ((instance: any): FunctionInstance);
```

Comment:
Why is this case needed? To assert that functionInstance is not `undefined`? Prefer an invariant in that case.